launchpod/src/test/java/com/google/launchpod/FormHandlerServletTest.java

```diff
@@ -0,0 +1,351 @@
+
+        Query query = new Query(USER_FEED);
+        Entity entity = ds.prepare(query).asSingleEntity();
+        thrown.expect(IOException.class);
+        thrown.expectMessage("There was no MP3 link inputted. Try again");
+    }
+
+    /**
+     * Asserts that doGet() returns correct XML string when given an entity ID.
+     * @return XML string 
+     * @throws IOException
+    */
+    @Test
+    public void doGetReturnsCorrectXmlString() throws IOException {
+        DatastoreService ds = DatastoreServiceFactory.getDatastoreService();
+        Entity entity = makeEntity(TEST_PODCAST_TITLE, TEST_MP3_LINK, TEST_TIMESTAMP, TEST_XML_STRING, TEST_EMAIL);
+        Entity entityTwo = makeEntity(TEST_PODCAST_TITLE, TEST_MP3_LINK, TEST_TIMESTAMP, TEST_XML_STRING, TEST_EMAIL);
+        String id = KeyFactory.keyToString(entity.getKey());
+        String idTwo = KeyFactory.keyToString(entityTwo.getKey());
+        ds.put(entity);
+        ds.put(entityTwo);
+
+        when(request.getParameter(ID)).thenReturn(id);
+
+        StringWriter stringWriter = new StringWriter();
+        PrintWriter writer = new PrintWriter(stringWriter);
+        when(response.getWriter()).thenReturn(writer);
+
+        servlet.doGet(request, response);
+
+        TEST_XML_STRING = xmlString(entity);
+
+        verify(response, atLeast(1)).setContentType("text/html");
+        writer.flush();
+        assertEquals(stringWriter.toString(), "<p>" + TEST_XML_STRING + "</p>");
+    }
+
+    /**
+     * Expects that doGet() returns an error message when an entity with request id does not exist in Datastore.
+     * @return invalid link message
+     * @throws IOException
+    */
+    @Test
+    public void doGetEntityNotFound() throws IOException {
+        DatastoreService ds = DatastoreServiceFactory.getDatastoreService();
+        Entity entity = makeEntity(TEST_PODCAST_TITLE, TEST_MP3_LINK, TEST_TIMESTAMP, TEST_XML_STRING, TEST_EMAIL);
+        Entity entityTwo = makeEntity(TEST_PODCAST_TITLE, TEST_MP3_LINK, TEST_TIMESTAMP, TEST_XML_STRING, TEST_EMAIL);
+        String id = KeyFactory.keyToString(entity.getKey());
+        String idTwo = KeyFactory.keyToString(entityTwo.getKey());
+        ds.put(entity);
+        ds.put(entityTwo);
+
+        when(request.getParameter(ID)).thenReturn(id);
+
+        StringWriter stringWriter = new StringWriter();
+        PrintWriter writer = new PrintWriter(stringWriter);
+        when(response.getWriter()).thenReturn(writer);
+
+        servlet.doGet(request, response);
+
+        verify(response, atLeast(1)).setContentType("text/html");
+        writer.flush();
+        assertEquals("<p>Sorry. This is not a valid link.</p>", stringWriter.toString());
+    }
+
+    /**
+     * Asserts that doGet() returns an error message when there are no entities in Datastore period.
+     * @throws IOException
+    */
+    @Test
+    public void doGetReturnsErrorMsgForNoEntitiesInDatastore() throws IOException {
```

Comment:
I don't think we need this test for this particular file. Since the user isn't logged in, we're telling the user that nobody has ever used our creation tool. It would make more sense to have this test in our login file later on.

Reply:
Makes sense- should I keep it in this PR since we might need it in the future or just delete it and rewrite it later?

Reply:
Yeah I would leave it and just add a todo.